provision-contest/ansible/EOC/README.md

```diff
@@ -0,0 +1,79 @@
+33. `Compare awards.json between primary and CDS`. Tag for just this action: `awards`. Note, only if CDS credentials are provided.
+36. `Verify that Primary and ... System Test`. Tag for just this action: `fetch`. Note, a manual commit is still needed and the `final_standings.html` is actually the static-scoreboard zip.
+
+
+## Setup
+To run this playbook a facts-file is required. An example of which can be seen below. Facts prefixed with:
+ - `dj_` are for interaction with a DOMjudge instance.
+ - `other_` are for interaction with another CLICS spec compliant CCS (commonly shadow if DOMjudge is primary, or vica versa)
+ - `cds_` are for interaction with the CDS (also CLICS compliant).
+
+Only the `repo`, `contest`, `contest_id`, and `dj_*` variables are required. When the `other_` or `cds_` facts are missing, steps which would interact with these systems are skipped.
+
+- `repo` must point to where all retrieved files should be stored, not simply the repo root!
+- `contest` must be the externalID.
+- `contest_id` must be the internalID. (cID)
+- `jd_loc` must point to the location of a [`jd` binary](https://github.com/josephburnett/jd/releases/tag/v1.6.1). Used for comparing diffs.
+
+```yaml
+repo: /home/mart/icpc/wf_repo
+contest: bapc2022
+contest_id: 2
+jd_loc: /home/mart/icpc/EOC/jd
+
+dj_url: https://judge.gehack.nl
+dj_url_api_suffix: api/v4
+dj_username: mart
+dj_password: REDACTED
+
+other_url: https://judge.gehack.nl
+other_url_api_suffix: api/v4
+other_username: mart
+other_password: REDACTED
+
+cds_url: https://cds.gehack.nl
+cds_url_api_suffix: api
+cds_username: admin
+cds_password: REDACTED
+```
+
+## Selectively running the playbook
+The playbook has multiple tags:
+ - `all`: runs all tasks
+ - `check`: Checks the consistency of both scoreboard.json and awards.json against the CDS and other CCS when configured.
+ - `scoreboard`: Verifies scoreboard.json against the CDS and other CCS when configured.
+ - `awards`: Verifies awards.json against the CDS and other CCS when configured.
+ - `results.tsv`: Downloads results.tsv from DOMjudge and stores it in the repo.
+ - `fetch`: Retrieves all files required by the EOC including results.tsv
+ - `dump`: Dumps all API endpoints, specifically:
+   - accounts
+   - awards
+   - balloons
+   - clarifications
+   - problems
+   - groups
+   - judgements
+   - languages
+   - organizations
+   - runs
+   - scoreboard
+   - teams
+
+### Known limitations
+ - The checks that verify whether the jsons are the same are *very* simple and I (Mart) have not yet found a reasonable way of actually checking them in a nice manner from ansible. Them matching would be a bigger red-flag than them having a difference. Still, the playbook fails when a difference is detected! To aid with (manual) verification the results of fetching the jsons is stored in `/tmp/[obj].[sys].json` with `[object]` either "awards" or "scoreboard" and `[sys]` either "dj", "cds", or "other".
```

Comment:
We should use the icpc tools contestutil for this

Reply:
If you have documentation for that for me. I'll make sure it works. :)

Reply:
```
./contestUtil.sh ScoreboardUtil ~/Desktop/dj.json ~/Desktop/cdsm.json # for scoreboard compare
```

For awards it doesn't exist yet... There is a way to compare the whole event feed.

@deboer-tim could we get an awards.json compare as a separate util?

Reply:
Implemented. awards.json still makes use of the borked jd method.